trader/trader.go

```diff
@@ -401,6 +401,8 @@ func (t *Trader) update() bool {
 			return false
 		}
 
+		t.metricsTracker.AddPruneOps(len(pruneOps))
```

Comment:
we shouldn't update the `metricsTracker` with this because the metrics tracker is only responsible to send metrics, not to accumulate metrics.

the `update()` function should return a struct like this:
```
{
success: bool
numPruneOps: int
numUpdateOpsDelete: int
numUpdateOpsUpdate: int
}
```

then in the caller, we are already sending `sendUpdateEvent` -- this should take in this new struct instead of taking in the `success` field which is currently a `bool`.

does this make sense?